src/api/controllers/authController.js

```diff
@@ -97,4 +98,190 @@ export default class Auth {
       return sendError(status.SERVER_ERROR, res, 'logout', 'Please try again, Thanks');
     }
   }
+
+  /**
+   * This function gets all the users that are stored in the User table
+   *
+   * @static
+   * @param {Object} req the request
+   * @param {Object} res the response to be sent
+   * @memberof users
+   * @returns {Object} res
+   */
+  static async getUsers(req, res) {
+    // default constants
+    const defaultLimit = 10;
+    const defaultOffset = 0;
+    const result = {};
+    const emptyUserArray = 0;
+
+    const limit = req.query.limit || defaultLimit;
+    const offset = req.query.offset || defaultOffset;
+    const users = await User.findAndCountAll({
+      offset,
+      limit
+    });
+
+    result.status = status.OK;
+    result.pagedUsers = paginateUser(users.count, users.rows, offset, limit);
+    result.users = users.rows;
+    if (users.rows.length > emptyUserArray) {
+      res.status(status.OK).json({
+        result
+      });
+    } else {
+      res.status(status.NOT_FOUND).json({
+        status: status.NOT_FOUND,
+        message: 'There is no more users at the moment.'
+      });
+    }
+  }
+
+  /**
+   * Update User
+   *
+   * @static
+   * @param {object} req - request body
+   * @param {object} res - response body
+   * @returns { object } - response
+   * @memberof Auth
+   */
+  static async updateUser(req, res) {
+    const { email } = req.params;
+    const {
+      firstName, lastName, username, password
+    } = req.body;
+    const salt = genSaltSync(parseFloat(env.hashRounds));
+    const hashedPassword = hashSync(password, salt);
+
+    const result = await User.update({
+      email,
+      firstName,
+      lastName,
+      username,
+      password: hashedPassword
+    },
+    {
+      where: {
+        email
+      }
+    });
+    return res.status(status.OK).json({
+      status: status.OK,
+      message: 'user updated successufully',
+      data: result
+    });
+  }
+
+  /**
+   * make User a modulator
+   *
+   * @static
+   * @param {object} req - request body
+   * @param {object} res - response body
+   * @returns { object } - response
+   * @memberof Auth
+   */
+  static async makeModulator(req, res) {
```

Comment:
@Ramadhan0 it's `moderator` not `modulator`

Reply:
@Ramadhan0 please, change `modulator` to `moderator` in all files where've you used `modulator`